### DISCUSSION TITLE
Question about Grain IDs when Creating a Tess File from a Raster File

### DISCUSSION THREAD

**Discussion by nippyawn** [neperfepx/neper#927] [Q&A] [answered]
Hello,

I am using Dream3D to convert 3D slice images into a raster format and then creating a tesr file. After that, I generated a tess file, followed by msh and inp files.

My question is regarding the grain IDs during this process. Are the grain IDs from the raster format carried over when creating the tess file, or are new random IDs assigned? I did not use the -transform resetcellid option when creating the tess file from the tesr file.

I followed the steps outlined in the tutorial below: https://neper.info/doc/tutorials/morpho_tesr_mesh.html

**Comment by rquey** [comment, marked as answer]
They are carried over, just as the crystal orientation (if defined). You can double-check by coloring both the tesr and tess by cell id (which is also the default):
```
neper -T <file.tesr> -datacellcol id -print tesr
neper -T <file.tess> -datacellcol id -print tess
```
